### ISSUE TITLE
swarm: Incorrect completion timestamp for completed tasks in task history

### ISSUE THREAD

**Issue by abronan** [docker/docker#25418]
docker **master**

Completion timestamp on task history/status does not make much sense, in the snippet below we can see that the tasks in the history were all completed `7 seconds` ago, which is incorrect:

```
user@node01:~# docker service ls
ID            NAME   REPLICAS  IMAGE  COMMAND
30memtfuxn6e  redis  5/5       redis

user@node01:~# docker service ps redis
ID                         NAME         IMAGE  NODE    DESIRED STATE  CURRENT STATE            ERROR
591kko02y4ekwgqu8onwq6zz9  redis.1      redis  node03  Running        Running 5 minutes ago    
ddod9ojf018upi48sknba7641  redis.2      redis  node02  Ready          Preparing 2 seconds ago  
bvfqbx93j1i0z0gvyfon53wxc   \_ redis.2  redis  node02  Shutdown       Complete 7 seconds ago   
8a9joi6fvfh66oavdnfc659iv   \_ redis.2  redis  node02  Shutdown       Complete 7 seconds ago   
8i03t6lc19if6en108bj6orob   \_ redis.2  redis  node02  Shutdown       Complete 7 seconds ago   
c9isnlj8mcc64xlw59tro3prr   \_ redis.2  redis  node02  Shutdown       Complete 7 seconds ago   
412sp6xod30ckmygmb1lsh90i  redis.3      redis  node01  Ready          Ready 1 seconds ago      
2rk7akdio7yveey1l4vg6j6gu   \_ redis.3  redis  node01  Shutdown       Complete 2 seconds ago   
b1kaz4sr56fnvizrxh29w51ar   \_ redis.3  redis  node01  Shutdown       Complete 7 seconds ago   
3nkgb8lrdzjrccx8x0hzy2x1e   \_ redis.3  redis  node01  Shutdown       Complete 7 seconds ago   
b5iqck9fur4xqd2kowo7itybr   \_ redis.3  redis  node01  Shutdown       Complete 7 seconds ago   
c74q8z0zkzu5wo5scdmnvr6qu  redis.4      redis  node01  Ready          Ready 1 seconds ago      
6qo374itvvqj047e24p44tzfa   \_ redis.4  redis  node02  Shutdown       Complete 2 seconds ago   
08y0x26j8jjbeqtu8jg7x8vj3   \_ redis.4  redis  node02  Shutdown       Complete 7 seconds ago   
8asvdk0q6rtj85e1dznec4fy5   \_ redis.4  redis  node02  Shutdown       Complete 7 seconds ago   
02cvnyxnh6kn63vc07ydek044   \_ redis.4  redis  node02  Shutdown       Complete 7 seconds ago   
0u35ngxxaa7o9jk5xw368216w  redis.5      redis  node02  Ready          Preparing 2 seconds ago  
116m3242rkv8foewx16gmms8a   \_ redis.5  redis  node01  Shutdown       Complete 7 seconds ago   
bk50f1vwlyuxr9r53ufrcxe0k   \_ redis.5  redis  node02  Shutdown       Complete 7 seconds ago   
a87ym0tjy5z79g8ul5q9z5xqf   \_ redis.5  redis  node02  Shutdown       Complete 7 seconds ago   
2cujhyab6aznc3p2d6xyi6g86   \_ redis.5  redis  node02  Shutdown       Complete 7 seconds ago  

```

**Comment by abronan** [comment]
docker/swarmkit#1336 solves the issue, we can close this one after vendoring a new swarmkit version.

**Comment by thaJeztah** [comment]
ping @aaronlehmann @stevvooe I see https://github.com/docker/swarmkit/pull/1336 is on a 1.12.1 milestone; was that change merged in this repository? Was it included in 1.12.1?

**Comment by aaronlehmann** [comment]
Yes, it has the `process/cherry-picked` label, so it was included in 1.12.1.
